src/ir/from_ast/build_ctx.rs

```diff
@@ -109,16 +137,12 @@ pub(super) struct BuildCtx<'ctx, 'prog> {
     /// Index for generating unique names
     name_idx: u32,
 }
-impl<'ctx, 'prog> BuildCtx<'ctx, 'prog> {
-    pub fn new(
-        ctx: &'ctx ir::Context,
-        comp: &'ctx mut ir::Component,
-        sigs: &'prog SigMap,
-    ) -> Self {
+
+impl<'prog> BuildCtx<'prog> {
+    pub fn new(comp: ir::Component) -> Self {
         Self {
-            ctx,
             comp,
-            sigs,
+            sigs: None,
```

Comment:
Why can this be constructed with no sigs?

Reply:
Because the sigs need the signatures of the components to be compiled first (to get the foreign portidxs) so there needs to be a period before `builder.sig()` is run where the signatures don't actually exist. We could use SigMap::default() instead which would probably be better.